lib/rom/sql/plugin/associates.rb

```diff
@@ -39,14 +35,33 @@ def initialize(*)
           # @overload SQL::Commands::Create#execute
           #
           # @api public
-          def execute(tuples, parent)
-            fk, pk = association[:key]
+          def execute(tuples, parent, assoc = association[:key])
+            input_tuples =
+              case assoc
+              when Array
+                fk, pk = assoc
 
-            input_tuples = with_input_tuples(tuples).map { |tuple|
-              tuple.merge(fk => parent.fetch(pk))
-            }
+                input_tuples = with_input_tuples(tuples).map { |tuple|
+                  tuple.merge(fk => parent.fetch(pk))
+                }
 
-            super(input_tuples)
+                super(input_tuples)
+              when Hash
+                new_tuples = super(tuples)
+
+                join_relation, keys = assoc.to_a[0]
```

Comment:
It's better to unwrap values once because they don't change
```ruby
when Hash
   super(tuples).tap do |new_tuples|
     join_relation, ((spk, sfk), (tfk, tpk)) = assoc.to_a[0]

     join_tuples = new_tuples.map do |tuple|
       { sfk => tuple.fetch(spk), 
         tfk => parent.fetch(tpk) }
     end
     
     join_relation.multi_insert(join_tuples)
   end
 end
```